catalog/app/containers/Auth/Provider.js

```diff
@@ -53,39 +52,38 @@ const useNotificationHandlers = () => {
 /**
  * Provider component for the authentication system.
  */
-export default RT.composeComponent(
-  'Auth.Provider',
-  RC.setPropTypes({
-    /**
-     * Determines on which actions to fire the check logic.
-     */
-    checkOn: PT.oneOfType([ActionPattern, PT.arrayOf(ActionPattern)]),
-    /**
-     * Storage instance used to persist tokens and user data.
-     */
-    storage: StorageShape.isRequired,
-    /**
-     * Expected API latency in seconds.
-     */
-    latency: PT.number,
-  }),
-  ({ children, checkOn, storage, latency = 20 }) => {
-    const reducerWithInit = React.useMemo(() => {
-      const init = fromJS(storage.load())
-        .filter(Boolean)
-        .update((s) =>
-          s.set('state', s.getIn(['user', 'current_user']) ? 'SIGNED_IN' : 'SIGNED_OUT'),
-        )
-      return withInitialState(init, reducer)
-    }, [])
-    useReducer(REDUX_KEY, reducerWithInit)
+export default function AuthProvider({
+  children,
+  /**
+   * Determines on which actions to fire the check logic.
+   */
+  checkOn, // oneOfType([ActionPattern, PT.arrayOf(ActionPattern)])
```

Comment:
i don't understand why we've commented out all of the type checks here

Reply:
they dont seem to help much and i plan to migrate over to typescript for compile-time type-checking in some near-ish future